pkg/controllermanager/deployer/deployer.go

```diff
@@ -357,9 +357,16 @@ func (deployer *Deployer) populateBasesAndLocalizations(sub *appsapi.Subscriptio
 			return fmt.Errorf("failed to populate Bases for Subscription %s: %v", klog.KObj(sub), err)
 		}
 
+		var baseName string
+		if len(allExistingBases) > 0 {
```

Comment:
This is WRONG!

This can NOT be the statement to identify whether we should use legacy base name. The corresponding clusters for `allExistingBases` may be probably different (or even totally different) from `sub.Status.BindingClusters`.